addons/binding/org.openhab.binding.ihc/src/main/java/org/openhab/binding/ihc/internal/ChannelUtils.java

```diff
@@ -0,0 +1,233 @@
+import org.eclipse.smarthome.core.thing.binding.builder.ChannelBuilder;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeUID;
+import org.openhab.binding.ihc.internal.config.ChannelParams;
+import org.openhab.binding.ihc.internal.handler.IhcHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NodeList;
+
+/**
+ * Generic methods related to openHAB channels.
+ *
+ * @author Pauli Anttila - Initial contribution
+ */
+public class ChannelUtils {
+    private static final Logger LOGGER = LoggerFactory.getLogger(IhcHandler.class);
+
+    public static Set<Integer> getAllChannelsResourceIds(Thing thing) {
+        Set<Integer> resourceIds = new HashSet<>();
+
+        thing.getChannels().forEach(c -> {
+            ChannelParams params = new ChannelParams(c);
+            if (params.getResourceId() != null && params.getResourceId() != 0) {
+                resourceIds.add(params.getResourceId());
+            }
+        });
+
+        return resourceIds;
+    }
+
+    public static Set<Integer> getAllTriggerChannelsResourceIds(Thing thing) {
+        Set<Integer> resourceIds = new HashSet<>();
+
+        thing.getChannels().forEach(c -> {
+            ChannelParams params = new ChannelParams(c);
+            if (params.getChannelTypeId() != null) {
+                switch (params.getChannelTypeId()) {
+                    case CHANNEL_TYPE_PUSH_BUTTON_TRIGGER:
+                        if (params.getResourceId() != null && params.getResourceId() != 0) {
+                            resourceIds.add(params.getResourceId());
+                        }
+                        break;
+                }
+            }
+        });
+        return resourceIds;
+    }
+
+    public static void addChannelsFromProjectFile(Thing thing, Document projectFile, List<Channel> thingChannels) {
+        if (projectFile != null) {
+            try {
+                NodeList nodes = projectFile.getElementsByTagName("product_dataline");
+
+                for (int i = 0; i < nodes.getLength(); i++) {
+                    Element node = (Element) nodes.item(i);
+                    addChannelsFromProjectFile(thing, node.getElementsByTagName("dataline_input"), "Switch", "input",
+                            CHANNEL_TYPE_SWITCH, thingChannels);
+                    addChannelsFromProjectFile(thing, node.getElementsByTagName("dataline_output"), "Switch", "output",
+                            CHANNEL_TYPE_SWITCH, thingChannels);
+                    addChannelsFromProjectFile(thing, node.getElementsByTagName("resource_temperature"), "Number",
+                            "temperature", CHANNEL_TYPE_NUMBER, thingChannels);
+                    addChannelsFromProjectFile(thing, node.getElementsByTagName("resource_humidity_level"), "Number",
+                            "humidity", CHANNEL_TYPE_NUMBER, thingChannels);
+                }
+            } catch (Exception e) {
```

Comment:
Please catch specific exception only (or `RuntimeException` if you want to catch unchecked exceptions.) Also check other places.